java-samples/corda5-negotiation-cordapp/contracts/src/main/java/com/r3/developers/samples/negotiation/contracts/ProposalAndTradeContract.java

```diff
@@ -0,0 +1,100 @@
+            @Override
+            public void verify(UtxoLedgerTransaction transaction){
+            Trade tradeStateOutput = transaction.getOutputStates(Trade.class).get(0);
+            Proposal proposalStateInputs =transaction.getInputStates(Proposal.class).get(0);
+
+            require(transaction.getInputStateAndRefs().size()==1,"There is exactly one input");
+            require(transaction.getInputStates(Proposal.class).size() == 1, "The single input is of type ProposalState");
+            require(transaction.getOutputTransactionStates().size()==1,"There is exactly one output");
+            require(transaction.getOutputStates(Trade.class).size()==1,"The single output is of type TradeState");
+            require(transaction.getCommands().size()==1,"There is exactly one command");
+
+            require( tradeStateOutput.getAmount()==proposalStateInputs.getAmount(),"The amount is unmodified in the output");
+            require(proposalStateInputs.getBuyer().toString().equals(tradeStateOutput.getBuyer().toString()),"The buyer is unmodified in the output");
+            require(proposalStateInputs.getSeller().toString().equals(tradeStateOutput.getSeller().toString()),"The seller is unmodified in the output");
+
+            require(transaction.getSignatories().contains(proposalStateInputs.getProposer().getLedgerKey()),"The proposer is a required signer");
+            require(transaction.getSignatories().contains(proposalStateInputs.getProposee().getLedgerKey()),"The proposee is a required signer");
+
+            }
+        }
+
+        public static class Modify implements NegotiationCommands {
+            @Override
+            public void verify(UtxoLedgerTransaction transaction){
+            Proposal proposalStateOutput = transaction.getOutputStates(Proposal.class).get(0);
+            Proposal proposalStateInputs =transaction.getInputStates(Proposal.class).get(0);
+
+            require(transaction.getInputStateAndRefs().size()==1,"There is exactly one input");
+            require(transaction.getInputStates(Proposal.class).size() == 1,"The single input is of type ProposalState");
+            require(transaction.getOutputTransactionStates().size()==1,"There is exactly one output");
+            require(transaction.getOutputStates(Proposal.class).size()==1,"The single output is of type ProposalState");
+            require(transaction.getCommands().size()==1,"There is exactly one command");
+
+            require(proposalStateOutput.getAmount() != proposalStateInputs.getAmount(),"The amount is modified in the output");
+            require(Objects.equals(proposalStateInputs.getBuyer().toString(), proposalStateOutput.getBuyer().toString()),"The buyer is unmodified in the output");
+            require(Objects.equals(proposalStateInputs.getSeller().toString(), proposalStateOutput.getSeller().toString()),"The seller is unmodified in the output");
+
+            //warn message over members
```

Comment:
comments are useful for explaining to cordapp developers the reason behind the next few lines of code or what it does.

it's unclear what this comment does, and why this merits a comment over the others.